example/scripts/lock-stack.js

```diff
@@ -3,48 +3,160 @@
 'use strict';
 
 /**
- * Functionality for locking stacks with expiration using the Mutex provided by
- * @cumulus/common. Stacks are locked based on the deployment name.
+ * Functionality for locking stacks with expiration and SHA tracking
  *
  * To use this, a DynamoDB table is required with the primary partition key being
  * a string - 'key' and no sort key. The table name should be set in LOCK_TABLE_NAME
  */
 
-const {
-  concurrency
-} = require('@cumulus/common');
-const aws = require('@cumulus/common/aws');
+const { aws } = require('@cumulus/common');
 
-const LOCK_TABLE_NAME = 'cumulus-int-test-lock';
-const STACK_EXPIRATION_MS = 120 * 60 * 1000; // 2 hours
+class CumulusLockError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = this.constructor.name;
```

Comment:
Is this ... necessary? Are you just trying to avoid "useless constructor" ESlint warnings? Separately, would it be good to maintain these error classes in `@cumulus/errors` and use that package here?

Reply:
Again, I'm not convinced that the CI scripting should push code that isn't *actually* something that belongs in core into core/errors.     I'm willing to be convinced I'm wrong.

Reply:
> Is this ... necessary? Are you just trying to avoid "useless constructor" ESlint warnings? 

Hmm, I think so, the createErrorType boilerplate/generation function  in cumulus/error is doing something similar when it sets the prototype name object, I presume for the same reason. 

Looking at it: 

> class CumulusLockErrorTest extends Error {
...   constructor(message) {
.....   super(message);
..... }}
undefined
> bar = new CumulusLockErrorTest('bar')
Error: bar
    at repl:1:7
    at ContextifyScript.Script.runInThisContext (vm.js:50:33)
    at REPLServer.defaultEval (repl.js:240:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:468:10)
    at emitOne (events.js:121:20)
    at REPLServer.emit (events.js:211:7)
    at REPLServer.Interface._onLine (readline.js:282:10)
    at REPLServer.Interface._line (readline.js:631:8)
> bar.name
'Error'

versus 

> class CumulusLockErrorTest2 extends Error {
... constructor(message) {
..... super(message);
..... this.name = this.constructor.name
..... }}
undefined
> bar = new CumulusLockErrorTest2('bar')
{ CumulusLockErrorTest2: bar
    at repl:1:7
    at Script.runInThisContext (vm.js:122:20)
    at REPLServer.defaultEval (repl.js:332:29)
    at bound (domain.js:402:14)
    at REPLServer.runBound [as eval] (domain.js:415:12)
    at REPLServer.onLine (repl.js:642:10)
    at REPLServer.emit (events.js:203:15)
    at REPLServer.EventEmitter.emit (domain.js:448:20)
    at REPLServer.Interface._onLine (readline.js:308:10)
    at REPLServer.Interface._line (readline.js:656:8) name: 'CumulusLockErrorTest2' }
> bar.name
'CumulusLockErrorTest2'
>


Or.... am I missing something more general?